Sources/XMTP/Messages/Signature.swift

```diff
@@ -63,11 +63,40 @@ extension Signature {
 	}
 
 	var rawData: Data {
-		if ecdsaCompact.bytes.isEmpty {
-			return walletEcdsaCompact.bytes + [UInt8(Int(walletEcdsaCompact.recovery))]
+		switch union {
+		case let .ecdsaCompact(ecdsa):
+			return ecdsa.bytes + [UInt8(Int(ecdsa.recovery))]
+		case let .walletEcdsaCompact(ecdsa):
+			return ecdsa.bytes + [UInt8(Int(ecdsa.recovery))]
+		case .none:
+			return Data()
 		}
+	}
 
-		return ecdsaCompact.bytes + [UInt8(Int(ecdsaCompact.recovery))]
+	mutating func ensureEcdsaSignature() {
```

Comment:
Is this still necessary after this change upstream? https://github.com/xmtp/xmtp-node-go/pull/193

Asking because I do see a lot of lets turning into vars which is totally fine if it's necessary but seams like it would be cleaner if it was fixed in the go repo!

Reply:
Good question, I didn't see that PR had landed so I'll pull it down and see how stuff works in the new world.

Reply:
@elisealix22 I was about to remove a bunch of those! Thanks for giving me the heads up about that PR landing.